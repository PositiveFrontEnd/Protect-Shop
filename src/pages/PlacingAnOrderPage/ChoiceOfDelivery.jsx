import React from 'react'
import ChoiceOfDelivF from '../../components/Form/ChoiceOfDeliv/ChoiceOfDelivF'

const ChoiceOfDelivery = () => {
  return (
   <>
   <ChoiceOfDelivF />
   </>
  )
}

export default ChoiceOfDelivery