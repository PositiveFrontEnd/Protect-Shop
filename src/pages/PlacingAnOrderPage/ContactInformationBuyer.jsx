import React from "react";
import ContactInfoF from "../../components/Form/ContactInfo/ContactInfoF";

const ContactInformationBuyer = () => {
  return (
    <>
      <ContactInfoF />
    </>
  );
};

export default ContactInformationBuyer;
