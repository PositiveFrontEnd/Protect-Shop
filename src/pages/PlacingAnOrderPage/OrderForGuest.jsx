// import React, { useEffect, useState } from "react";
// import { useDispatch, useSelector } from "react-redux";
// import { actionCreateNewOrderNotLoggerCustomer, actionInfoOrderForGuest, actionProductsOrderForGuest } from "../../store/orderSlice";
// import { actionGetOneProduct } from "../../store/productsSlice";
// import { selectorDeliveryForOrderGuest, selectorGuestBasket, selectorInfoForOrderGuest, selectorOrderFormData , selectorProductsForOrderGuest } from "../../store/selectors";

// const OrderForGuest = () => {
//     const dispatch = useDispatch()
//     const guestBasket = useSelector(selectorGuestBasket)
//     const [productsOrderForGuest, setProductsOrderForGuest] = useState([])
//     const orderGuestInfo = useSelector(selectorOrderFormData )
//     useEffect(() => {
//         const fetchData = async() => {
//             try {
//                 const order = [];
//                 for (const key of Object.keys(guestBasket)) {
//                     const item = guestBasket[key];
//                     if (item.payload) {
//                         const data = await dispatch(actionGetOneProduct(item.payload));
//                         console.log(data);
//                         order.push({
//                             cartQuantity: item.counter,
//                             product: data,
//                         });
//                     }
//                 }
//                 setProductsOrderForGuest(order)
//                 console.log(order)
//             }
//             catch {
//                 console.log('error')
//             }
//         }
//         fetchData()
//     }, [dispatch, guestBasket])

//     const order = {
//         products: productsOrderForGuest,
//         ...orderGuestInfo
//     }
//     dispatch(actionProductsOrderForGuest(order));

//     console.log(order)

//     return (
//         <p> productsOrderForGuest</p>
//     )
// }
// export default OrderForGuest