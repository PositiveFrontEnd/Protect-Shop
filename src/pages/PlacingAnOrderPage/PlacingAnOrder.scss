.placing__order__box {
  display: flex;
  align-items: center;
  justify-content: center;
  flex-direction: column;
  margin-top: 35px;
  margin-bottom: 23px;

  h3 {
    color: var(--black);
    font-weight: 400;
    font-size: 18px;
    line-height: 25px;
    font-family: "Noto Sans";
    padding: 8px 20px;
  }
  .placing__order__link__box::before {
    content: "";
    display: block;
    border-bottom: 4px solid var(--inactive);
    min-width: 328px;
    position: absolute;
    top: 210px;
  }
  .placing__order__link__box {
    display: flex;
    align-items: center;
    justify-content: center;
    padding: 8px 0;
    gap: 10px;
    margin-bottom: 24px;

    a {
      color: var(--black);
      font-weight: 400;
      font-size: 12px;
      line-height: 17px;
      width: 160px;
    }
    a.active {
      color: var(--black);
      font-weight: 600;
      font-size: 12px;
      line-height: 17px;
    }
    a.active::after {
      content: "";
      display: block;
      border-bottom: 4px solid var(--black);
      position: relative;
      top: 16px;
    }
  }
  @media (width >= 768px) {
    max-width: 900px;

    .placing__order__link__box::before {
      top: 243px;
    }
    .placing__order__link__box {
      width: 100%;
      display: flex;
      align-items: center;
      justify-content: flex-start;
    }
  }
  @media (width>=1024px) {
    max-width: 1440px;
    .placing__order__link__box::before {
      top: 243px;
    }
    .placing__order__link__box::before {
      min-width: 530px;
      top: 253px;
    }
    .placing__order__link__box {
      a {
        color: var(--black);
        font-weight: 400;
        font-size: 16px;
        line-height: 22px;
        width: 260px;
      }
      a.active {
        color: var(--black);
        font-weight: 600;
        font-size: 16px;
        line-height: 22px;
      }
    }
  }
}
