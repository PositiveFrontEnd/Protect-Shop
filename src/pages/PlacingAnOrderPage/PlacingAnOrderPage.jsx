import React from "react";
import { NavLink, Outlet, useLocation } from "react-router-dom";
import "./PlacingAnOrder.scss";

const PlacingAnOrderPage = () => {
  const location = useLocation();

  const isNavLinkActive = (to) => {
    return location.pathname.includes(to);
  }
  return (
    <>
      <section className="container placing__order__box">
        <h3>PLACING AN ORDER</h3>
        <div className="placing__order__link__box">
          <NavLink to="contact_information" className={`${isNavLinkActive("/contact_information") ? "active" : ""}`}>
            1. CONTACT INFORMATION OF THE BUYER
          </NavLink>
          <NavLink to="choice_of_delivery" className={`${isNavLinkActive("/choice_of_delivery") ? "active" : ""}`}>
            2. CHOICE OF DELIVERY AND PAYMENT METHODS
          </NavLink>
        </div>
        <Outlet />
      </section>
    </>
  );
};

export default PlacingAnOrderPage;
