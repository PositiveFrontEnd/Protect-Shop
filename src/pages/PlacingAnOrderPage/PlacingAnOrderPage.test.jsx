import React from "react";
import { render } from "@testing-library/react";
import { Provider } from 'react-redux';
import PlacingAnOrderPage from "./PlacingAnOrderPage";
import { BrowserRouter } from "react-router-dom";
import configureStore from "redux-mock-store";

describe("Тестування компонента  PlacingAnOrderPage", () => {
    test("Знімок", () => {
        const mockStore = configureStore();
        const initialState = {

        };

        const store = mockStore(initialState);
        const placingAnOrderPage = render(
            <Provider store={store}>
                <BrowserRouter>
                    <PlacingAnOrderPage />
                </BrowserRouter>
            </Provider>
        );

        expect(placingAnOrderPage).toMatchSnapshot();
    });
});
