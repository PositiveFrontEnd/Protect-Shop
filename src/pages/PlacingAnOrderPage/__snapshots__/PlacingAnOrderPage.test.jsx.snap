// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Тестування компонента  PlacingAnOrderPage > Знімок 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <section
        class="container placing__order__box"
      >
        <h3>
          PLACING AN ORDER
        </h3>
        <div
          class="placing__order__link__box"
        >
          <a
            class=""
            href="/contact_information"
          >
            1. CONTACT INFORMATION OF THE BUYER
          </a>
          <a
            class=""
            href="/choice_of_delivery"
          >
            2. CHOICE OF DELIVERY AND PAYMENT METHODS
          </a>
        </div>
      </section>
    </div>
  </body>,
  "container": <div>
    <section
      class="container placing__order__box"
    >
      <h3>
        PLACING AN ORDER
      </h3>
      <div
        class="placing__order__link__box"
      >
        <a
          class=""
          href="/contact_information"
        >
          1. CONTACT INFORMATION OF THE BUYER
        </a>
        <a
          class=""
          href="/choice_of_delivery"
        >
          2. CHOICE OF DELIVERY AND PAYMENT METHODS
        </a>
      </div>
    </section>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
